### ISSUE TITLE
plain.html is not consistent with deployed index.html

### ISSUE THREAD

**Issue by Josh-Cena** [mdn/yari#11396]
### Summary

The output plain.html is not consistent with index.html, in particular the anchors are different.

### URL

https://developer.mozilla.org/en-US/docs/Web/CSS/Value_definition_syntax#question_mark

### Reproduction steps

1. Run `yarn build files/en-us/web/css/value_definition_syntax/index.md --nohtml`
2. Check `build/en-us/docs/web/css/value_definition_syntax/plain.html`; search for "question_mark"
3. See `<h3 id="question_mark_">`

### Expected behavior

The anchor is `question_mark`, because that's what the deployed page has (and what `yarn start` displays)

### Actual behavior

The anchor is `question_mark_`

### Device

Desktop

### Browser

Chrome

### Browser version

Stable

### Operating system

Mac OS

### Anything else?

Apologies in advance if this is something that's changed very recently. My Yari version is the latest.

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/orgs/mdn/discussions/categories/platform).

**Comment by caugner** [comment]
We don't make any guarantees about the contents of those `plain.html` files, as these are only used internally for our AI Help index. We really shouldn't write them as part of a regular ` yarn build`.

**Comment by Josh-Cena** [comment]
Since generating index.html is deprecated, what do we use to get the output HTML, then?

**Comment by caugner** [comment]
Rendering `index.html` as part of the build is deprecated, because it has been split to a separate `yarn render:html` command. Let me know if that helps.

**Comment by Josh-Cena** [comment]
Could you add that to content's `scripts`? I'm not sure if this is distributed in the Yari CLI.

Also, the index.html is much bigger than plain.html and hard to parse. Having a stable plain.html is still much appreciated, but it's fine if you want to treat that as an implementation detail.

**Comment by shreewiser** [comment]
Hi

**Comment by LeoMcA** [comment]
I'm intrigued that the plain.html files don't have the same ids as index.html, we should probably fix that in its own right for AI Help.

Otherwise, @Josh-Cena can you help me understand your workflow here? Why are you building and manually parsing the HTML files, rather than running `yarn start` and viewing the output live-built in your browser? What's the sequence of commands + actions you're (ideally) running?

**Comment by Josh-Cena** [comment]
@LeoMcA: I'm parsing plain.html as part of my link checker that detects broken anchor links. I run it on build output so it can handle links generated by macros.

This bug doesn't block me now because I have since switched to parse index.json instead, but I think this is still nice-to-have for Yari.

**Comment by caugner** [comment]
Our new [rari](https://github.com/mdn/rari) build system no longer generates these `plain.html`, so I'm going to close this as won't fix.
